betty/cropper/models.py

```diff
@@ -282,16 +282,20 @@ def clear_crops(self, ratios=None):
             ratios.append("original")
 
         for ratio_slug in ratios:
-            ratio_path = os.path.join(self.path(), ratio_slug)
-            if os.path.exists(ratio_path):
-                if settings.BETTY_CACHE_FLUSHER:
-                    for crop in os.listdir(ratio_path):
-                        width, format = crop.split(".")
-                        ratio = os.path.basename(ratio_path)
-                        full_url = self.get_absolute_url(ratio=ratio, width=width, format=format)
-                        settings.BETTY_CACHE_FLUSHER(full_url)
-
-                shutil.rmtree(ratio_path)
+            if settings.BETTY_CACHE_FLUSHER:
+                # Since might now know which formats to flush (since maybe not saving crops to
+                # disk), need to flush all possible crops.
+                # TODO: BETTY_CACHE_FLUSHER should support wildcards
+                for width in settings.BETTY_WIDTHS:
+                    for format in ["png", "jpg"]:
+                        url = self.get_absolute_url(ratio=ratio_slug, width=width, format=format)
+                        settings.BETTY_CACHE_FLUSHER(url)
+
+            # Delete entire crop ratio directory
+            if settings.BETTY_SAVE_CROPS:
```

Comment:
The idea was to you only have to delete the crops directory if you're in "legacy save crops mode". If not, then don't delete it. This is how crops are cleared, b/c then NGINX will have pass next request to betty instead of directly serving cached file on disk.